riocli/deployment/execute.py

```diff
@@ -33,33 +34,58 @@
     help_headers_color=Colors.YELLOW,
     help_options_color=Colors.GREEN,
 )
-@click.option('--component', 'component_name', default=None,
-              help='Name of the component in the deployment')
+@click.option('--user', default='root')
+@click.option('--shell', default='/bin/bash')
 @click.option('--exec', 'exec_name', default=None,
               help='Name of a executable in the component')
 @click.argument('deployment-name', type=str)
 @click.argument('command', nargs=-1)
-# @name_to_guid
 def execute_command(
-        component_name: str,
+        user: str,
+        shell: str,
         exec_name: str,
         deployment_name: str,
-        deployment_guid: str,
         command: typing.List[str]
 ) -> None:
     """
     Execute commands on cloud deployment
     """
     try:
-        comp_id, exec_id, pod_name = select_details(deployment_guid, component_name, exec_name)
+        client = new_v2_client()
+
+        deployment = client.get_deployment(deployment_name)
+        if not deployment:
+            click.secho(f'Deployment `{deployment_name}` not found', fg=Colors.RED)
+            raise SystemExit(1)
+        
+        if deployment.status.status != Status.RUNNING:
+            click.secho(f'Deployment `{deployment_name}` is not in Running state', fg=Colors.RED)
+            raise SystemExit(1)
+        
+        if deployment.spec.runtime != 'device':
+            click.secho(f'Execution on cloud deployments is not supported', fg=Colors.RED)
+            raise SystemExit(1)
+        
```

Comment:
todo: If no `exec_name` is given and the deployment has only one executable, use that executable in the `run_on_device` command. If the deployment has multiple executables and no `exec_name` is given, then `show_selection` and let the user select one.

You can move the logic you added in `run_on_device` here.